src/collection-view.js

```diff
@@ -644,18 +693,15 @@ const CollectionView = Backbone.View.extend({
   },
 
   // Destroy the child views that this collection view is holding on to, if any
-  _destroyChildren({checkEmpty} = {}) {
-    this.triggerMethod('before:destroy:children', this);
-    const shouldCheckEmpty = checkEmpty !== false;
+  _destroyChildren(options) {
     const childViews = this.children.map(_.identity);
 
-    this.children.each(_.bind(this.removeChildView, this));
-
-    if (shouldCheckEmpty) {
```

Comment:
These are the failing test when I move the check back.

1) collectionview - emptyView when the last item is removed from a collection should append the html for the emptyView:
     AssertionError: expected [object Object] to have html '<span>empty</span>' but found ''
      at Context.<anonymous> (collection-view.empty-view.spec.js:155:47)

  2) collectionview - emptyView when the last item is removed from a collection should call "onBeforeRenderEmpty":
     AssertionError: expected onBeforeRenderEmpty to have been called exactly once, but it was called 0 times
      at Context.<anonymous> (collection-view.empty-view.spec.js:159:7)

  3) collectionview - emptyView when the last item is removed from a collection should call "onRenderEmpty":
     AssertionError: expected onRenderEmpty to have been called exactly once, but it was called 0 times
      at Context.<anonymous> (collection-view.empty-view.spec.js:165:7)

  4) collectionview - emptyView when the last item is removed from a collection should reference each of the rendered view items:
     AssertionError: expected { Object (_views, _indexByModel, ...) } to have a length of 1 but got 0
      at Context.<anonymous> (collection-view.empty-view.spec.js:171:52)

  5) collection view - filter when rendering a collection where some models pass the filter when all models passing the filter are removed from the collection should show the EmptyView:
     AssertionError: expected [object Object] to contain text 'empty' but found ''
      at Context.<anonymous> (collection-view.filter.spec.js:134:52)

  6) collection view when providing a custom render that adds children, without a collection object to use, and removing a child should show the empty view:

      AssertionError: expected 0 to equal 1
      + expected - actual

      -0
      +1
      
      at Context.<anonymous> (collection-view.spec.js:631:60)

  7) collection view when providing a custom render that adds children, without a collection object to use, and removing a child should call "onBeforeRenderEmpty":
     AssertionError: expected onBeforeRenderEmpty to have been called at least once, but it was never called
      at Context.<anonymous> (collection-view.spec.js:635:7)

  8) collection view when providing a custom render that adds children, without a collection object to use, and removing a child should call "onRenderEmpty":
     AssertionError: expected onRenderEmpty to have been called at least once, but it was never called
      at Context.<anonymous> (collection-view.spec.js:639:7)

I'll take a look some more some time in the week.